packages/eslint-plugin-gestalt/src/eslintASTFixers.js

```diff
@@ -74,13 +74,59 @@ export const renameTagFixer: RenameTagFixerType = ({
   tagName,
 }) => {
   return [elementNode.openingElement, elementNode.closingElement]
-    .map((node) =>
-      node
-        ? fixer.replaceText(
+    .map((node) => {
+      // $FlowFixMe[incompatible-type] Flow is not detecting the method filter(Boolean)
+      if (!node) return false;
+
+      return fixer.replaceText(
+        node,
+        getTextNodeFromSourceCode({ context, elementNode: node }).replace(
+          tagName,
+          newComponentName,
+        ),
+      );
+    })
+    .filter(Boolean);
+};
+
+type RenameTagWithPropsFixerType = ({|
+  additionalPropsString: string,
+  context: GenericNode,
+  elementNode: GenericNode,
+  fixer: GenericNode,
+  newComponentName: string,
+  tagName: string,
+|}) => $ReadOnlyArray<GenericNode>;
+
+/** This function is a more complex version of renameTagFixer. It has the same tag replacement functionality, but it also rebuild the props in the opening tag to include a new prop. The new prop must be formatted as a string, p.e. `as="article"`
```

Comment:
Does this handle a component that already contains props, and we're adding another prop to it? If so, does it make more sense to keep the props as objects and only stringify as part of the return of this function? Or is it easier to just deal with strings the whole time?

Reply:
The API is basically working with Text
![image](https://user-images.githubusercontent.com/10593890/129113926-6b5e43d4-a297-413a-8ca3-e8875c909f29.png)

Reply:
There might be better solutions, I just found this one pretty straightforward

Reply:
What happened here is that I had to change both tags AND props. And it cannot be done in different fixers because fixers cannot overlap "fixing objects must not be overlapped". So I found it easier to build the opening/closing tags.